### PR TITLE
Improvements to auth flow

### PR DESCRIPTION
**Pull request by carteryu**
Items:

1. Route the back buttons and remove (comment out) unnecessary buttons [done]
2. Refactor profile screen + fix errors in flair data [done]
3. Logout should take u to landing page [done]
4. Make sure the "..." state is not clickable in the next button (most prominently on username select) [done]
5. Now actually pass in user data thru everything [done]

**The way it works now is there's only one DB write at the end of the signup flow. We save every step in async storage

### CODE REVIEW

File: civilapp/auth/AuthLoadingScreen.js [hunk 1]
```diff
@@ -148,6 +148,7 @@ export default class AuthLoadingScreen extends Component {
   //TODO: Consider exporting this function to be used in App initialization
   initializeUserData = async () => {
     try {
+      //NOTE: if your app doesn't load due to your user being deleted in cognito/dynamo, you should comment this part out
```
Comment (EphemeralX) [inline civilapp/auth/AuthLoadingScreen.js]: FYI, I updated aws-amplify to 1.2.3, from 1.1.36 and this was fixed.

File: civilapp/auth/FlairSetupScreen.js [hunk 2]
```diff
@@ -41,37 +43,31 @@ class FlairSetupScreen extends Component {
...
   }
 
   updateFlair = async flair => {
+    const phoneNumber = await AsyncStorage.getItem('PHONE_NUMBER')
+    const avatar = await AsyncStorage.getItem('AVATAR')
     const flairDetails = {
       id: this.state.userId,
+      phoneNumber,
+      //TODO: fix naming here depending on how we're storing the avatar
+      profileImageUrl: avatar,
```
Comment (EphemeralX) [inline civilapp/auth/FlairSetupScreen.js]: Just want to double check; if phoneNumber and avatar is undefined/null from AsyncStorage, this still works cleanly.

File: civilapp/auth/FlairSetupScreen.js [hunk 3]
```diff
@@ -84,54 +80,51 @@ class FlairSetupScreen extends Component {
...
+    />
   )
 
   render() {
     return (
       <RetroBackground>
         <TouchableOpacity
           onPress={() => {
-            this.props.navigation.navigate('ProfileSelectScreen')
+            this.props.navigation.navigate('Profile')
```
Comment (EphemeralX) [inline civilapp/auth/FlairSetupScreen.js]: If we're calling `ProfileSelectScreen` by `Profile`, what are we going to call the actual profile pages later on?  Maybe `ProfileSelect` is better here if we want to reserve`Profile` for the actual profile screen.

**Comment by rauchp** [comment]
you lucky I couldn't nit this, punk

**Comment by carteryu** [comment]
> you lucky I couldn't nit this, punk

LOL sorry man cody wanted this merged so he can test xD

**Pull request merged**